Replace gpdb-specific dbid detection in the pg_rewind

InnerLife0 opened the pull request:
Replace gpdb-specific dbid detection in the pg_rewind

pg_rewind util has gpdb-specific [logic](https://github.com/arenadata/gpdb/blob/eeee4d43b7179a69cd7ea4234736d214d4dffaf7/src/bin/pg_rewind/pg_rewind.c#L253) to detect current dbid. It [runs](https://github.com/arenadata/gpdb/blob/eeee4d43b7179a69cd7ea4234736d214d4dffaf7/src/bin/pg_rewind/pg_rewind.c#L914)
`postgres` which parses `postgresql.conf` and all included `.conf` files. Among
others, this file includes `internal.auto.conf` file which contains gp_dbid
parameter. There is no need to parse the whole `postgresql.conf` to get just
gp_dbid. More, this file may be zeroed by unsuccessful pg_rewind sync or may be
moved somewhere. The only needed file to run specific logic is
`internal.auto.conf`, which is [not synced](https://github.com/arenadata/gpdb/blob/eeee4d43b7179a69cd7ea4234736d214d4dffaf7/src/bin/pg_rewind/filemap.c#L117), so the solution is to pass it as
config to `postgres`.
New test file was created to show solution works when `postgresql.conf` is
zeroed.

BenderArenadata commented:
Allure report https://allure.adsw.io/launch/72606

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1492505

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1492507

BenderArenadata commented:
Failed job Regression tests with Postgres on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1492494

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1492496

BenderArenadata commented:
Failed job Regression tests with Postgres on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1492495

Commit:
Replace gpdb-specific dbid detection in the pg_rewind

pg_rewind util has gpdb-specific logic to detect current dbid. It runs
`postgres` which parses `postgresql.conf` and all included `.conf` files. Among
others, this file includes `internal.auto.conf` file which contains gp_dbid
parameter. There is no need to parse the whole `postgresql.conf` to get just
gp_dbid. More, this file may be zeroed by unsuccessful pg_rewind sync or may be
moved somewhere. The only needed file to run specific logic is
`internal.auto.conf`, which is not synced, so the solution is to pass it as
config to `postgres`.
New test file was created to show solution works when `postgresql.conf` is
zeroed.
